lib/IDE/CompletionLookup.cpp

```diff
@@ -1669,6 +1669,27 @@ void CompletionLookup::addSubscriptCall(const SubscriptDecl *SD,
   addTypeAnnotation(Builder, resultTy, SD->getGenericSignatureOfContext());
 }
 
+static StringRef getTypeAnnotationString(const NominalTypeDecl *NTD,
+                                         SmallVectorImpl<char> &stash) {
+  SmallVector<StringRef, 1> attrRoleStrs;
+  if (NTD->getAttrs().hasAttribute<PropertyWrapperAttr>())
+    attrRoleStrs.push_back("Property Wrapper");
+  if (NTD->getAttrs().hasAttribute<ResultBuilderAttr>())
+    attrRoleStrs.push_back("Result Builder");
+  if (NTD->isGlobalActor())
+    attrRoleStrs.push_back("Global Actor");
+
+  if (attrRoleStrs.empty())
+    return StringRef();
+  if (attrRoleStrs.size() == 0)
+    return attrRoleStrs[0];
+
+  stash.clear();
```

Comment:
Instead of clearing, would it be better to assert that it's empty? As otherwise, callers relying on a StringRef may be surprised that calling again could potentially empty it out.